core/server/worker/src/main/java/alluxio/worker/grpc/BlockWriteHandler.java

```diff
@@ -54,6 +62,7 @@
     super(responseObserver, userInfo);
     mWorker = blockWorker;
     mDomainSocketEnabled = domainSocketEnabled;
+    mRpcCompleted = false;
```

Comment:
It looks you can use `mContext.isDoneUnsafe()` as a completion flag instead of introducing a new flag.